lib/Stan_math_code_gen.ml

```diff
@@ -20,15 +36,17 @@ let%expect_test "emit function type raises" =
   | exception e -> print_s [%sexp (e : exn)] ) ;
   [%expect {| ((x UMathLibraryFunction) "not implemented yet") |}]
 
-let pp_call ppf (name, pp_args, args) =
-  pf ppf "%s(@[<hov>%a@])" name pp_args args
+let pp_call ppf (name, pp_arg, args) =
+  pf ppf "%s(@[<hov>%a@])" name (list ~sep:comma pp_arg) args
+
+let pp_call_str ppf (name, args) = pp_call ppf (name, string, args)
 
 let rec pp_expr ppf s =
   match s with
   | Var s -> string ppf s
-  | Lit (Str, s) -> pf ppf "\"%s\"" s
+  | Lit (Str, s) -> pf ppf "%S" s
   | Lit (_, s) -> string ppf s
-  | FunApp (fname, args) -> pp_call ppf (fname, list ~sep:comma pp_expr, args)
+  | FunApp (fname, args) -> pp_call ppf (fname, pp_expr, args)
   | BinOp (e1, op, e2) ->
       pf ppf "%a %s %a" pp_expr e1 (Operators.operator_name op) pp_expr e2
   | TernaryIf (cond, ifb, elseb) ->
```

Comment:
This all seems fine with me. The only one I cannot comment on is the Indexed expressions. Have you checked it does the same as the old code gen?

Reply:
it does for at least some cases. Haven't done much comprehensive testing in the module yet.